sdk/lib/_internal/js_runtime/lib/js_mirrors.dart

```diff
@@ -589,8 +587,18 @@ InstanceMirror reflect(Object reflectee) {
   }
 }
 
-TypeMirror reflectType(Type key) {
-  return reflectClassByMangledName(getMangledTypeName(key));
+TypeMirror reflectType(Type key, [List<Type> typeArguments]) {
+  String mangledName = getMangledTypeName(key);
+  if (typeArguments != null) {
+    if (!typeArguments.every((_) => _ is Type)) {
```

Comment:
Test that it's actually a `TypeImpl` and not just a user-implemented `Type` class.